assimilation_code/modules/assimilation/algorithm_info_mod.f90

```diff
@@ -128,107 +290,198 @@ subroutine probit_dist_info(kind, is_state, is_inflation, dist_type, &
-      stop
-   endif
+
+      ! Comparing the dist_type in string format to list of potential dist_types
+      dist_type_string = qcf_table_data(QTY_loc(1))%probit_extended_state%dist_type
+      call to_upper(dist_type_string)
+      dist_type_loc = findloc(possible_dist_types, trim(dist_type_string))
+
+      if (dist_type_loc(1) == 0) then
+         write(errstring, *) 'Invalid dist_type: ', trim(dist_type_string)
+         call error_handler(E_ERR, 'probit_dist_info:', errstring, source)
+
+      else
+         dist_type = possible_dist_type_ints(dist_type_loc(1))
+      endif
+
+      bounded_below = qcf_table_data(QTY_loc(1))%probit_extended_state%bounded_below
+      bounded_above = qcf_table_data(QTY_loc(1))%probit_extended_state%bounded_above
+      lower_bound = qcf_table_data(QTY_loc(1))%probit_extended_state%lower_bound
+      upper_bound = qcf_table_data(QTY_loc(1))%probit_extended_state%upper_bound
+
 endif
 
 end subroutine probit_dist_info
 
 !------------------------------------------------------------------------
 
 
-subroutine obs_inc_info(obs_kind, filter_kind, rectangular_quadrature, gaussian_likelihood_tails, &
+subroutine obs_inc_info(obs_qty, filter_kind, rectangular_quadrature, gaussian_likelihood_tails, &
    sort_obs_inc, spread_restoration, bounded_below, bounded_above, lower_bound, upper_bound)
 
-integer,  intent(in)  :: obs_kind
+integer,  intent(in)  :: obs_qty
 integer,  intent(inout) :: filter_kind
 logical,  intent(inout) :: rectangular_quadrature, gaussian_likelihood_tails
 logical,  intent(inout) :: sort_obs_inc
 logical,  intent(inout) :: spread_restoration
 logical,  intent(inout) :: bounded_below, bounded_above
 real(r8), intent(inout) :: lower_bound,  upper_bound
 
+integer :: QTY_loc(1)
+character(len=129) :: qty_name
+
+integer :: filter_kind_loc(1)
+character(len=129), dimension(5) :: possible_filter_kinds
+integer, dimension(5) :: possible_filter_kind_ints
+character(len=129) :: filter_kind_string
+
 ! The information arguments are all intent (inout). This means that if they are not set
 ! here, they retain the default values from the assim_tools_mod namelist. Bounds don't exist 
 ! in that namelist, so default values are set in assim_tools_mod just before the call to here.
 
 ! Temporary approach for setting the details of how to assimilate this observation
 ! This example is designed to reproduce the squared forward operator results from paper
 
-
-! Set the observation increment details for each type of quantity
-if(obs_kind == QTY_STATE_VARIABLE) then
+! Fill arrays with possible filter_kind strings and corresponding ints
+possible_filter_kinds(1) = 'EAKF'
+possible_filter_kinds(2) = 'ENKF'
+possible_filter_kinds(3) = 'UNBOUNDED_RHF'
+possible_filter_kinds(4) = 'GAMMA_FILTER'
+possible_filter_kinds(5) = 'BOUNDED_NORMAL_RHF'
+
+possible_filter_kind_ints(1) = 1
+possible_filter_kind_ints(2) = 2
+possible_filter_kind_ints(3) = 8
+possible_filter_kind_ints(4) = 11
+possible_filter_kind_ints(5) = 101
```

Comment:
Definite change: 

These filter_kind parameters are already defined in this module 
https://github.com/NCAR/DART/blob/a4723e7b706dd4be054e0e103de9703514ebda8a/assimilation_code/modules/assimilation/algorithm_info_mod.f90#L30-L38

Don't redefine them here. 

Convert the strings to the parameter value when reading the table, rather that doing the string comparison everytime.

Reply:
@hkershaw-brown I made this suggested change for both the dist_type and filter_kind. They are both now converted from string to integer in read_qcf_table and are set to integers in the types.